commit ce486ce9963438283836ed2224b0f0656271f99d
Author: Cameron Nemo <cnemo@tutanota.com>
Date:   Sat Aug 15 13:13:31 2020 -0700

    arm64: dts: rockchip: enable rk3328-rock64 usb3 nodes
    
    Enable USB3 nodes for the rk3328-based PINE Rock64 board.
    
    Signed-off-by: Heiko Stuebner <heiko@sntech.de>
    Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts b/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
index 86cfb5c50a94..0cbf59efcef0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-rock64.dts
@@ -384,6 +384,15 @@ &usb20_otg {
 	status = "okay";
 };
 
+&usbdrd3 {
+	status = "okay";
+};
+
+&usbdrd_dwc3 {
+	dr_mode = "host";
+	status = "okay";
+};
+
 &usb_host0_ehci {
 	status = "okay";
 };

commit 6266cc1763aa38153a4f5b69fad2eaa9259973e8
Author: Cameron Nemo <cnemo@tutanota.com>
Date:   Sat Aug 15 13:12:02 2020 -0700

    arm64: dts: rockchip: rk3328 usb3 controller node
    
    RK3328 SoCs have one USB 3.0 OTG controller which uses DWC_USB3
    core's general architecture. It can act as static xHCI host
    controller, static device controller, USB 3.0/2.0 OTG basing
    on ID of USB3.0 PHY.
    
    Signed-off-by: William Wu <william.wu@rock-chips.com>
    Signed-off-by: Heiko Stuebner <heiko@sntech.de>
    Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

diff --git a/arch/arm64/boot/dts/rockchip/rk3328.dtsi b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
index bbdb19a3e85d..27e86bf06d3e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3328.dtsi
@@ -983,6 +983,33 @@ usb_host0_ohci: usb@ff5d0000 {
 		status = "disabled";
 	};
 
+	usbdrd3: usb@ff600000 {
+		compatible = "rockchip,rk3328-dwc3";
+		clocks = <&cru SCLK_USB3OTG_REF>, <&cru SCLK_USB3OTG_SUSPEND>,
+			 <&cru ACLK_USB3OTG>;
+		clock-names = "ref_clk", "suspend_clk",
+			      "bus_clk";
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+		status = "disabled";
+
+		usbdrd_dwc3: dwc3@ff600000 {
+			compatible = "snps,dwc3";
+			reg = <0x0 0xff600000 0x0 0x100000>;
+			interrupts = <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>;
+			dr_mode = "otg";
+			phy_type = "utmi_wide";
+			snps,dis_enblslpm_quirk;
+			snps,dis-u2-freeclk-exists-quirk;
+			snps,dis_u2_susphy_quirk;
+			snps,dis_u3_susphy_quirk;
+			snps,dis-del-phy-power-chg-quirk;
+			snps,dis-tx-ipgap-linecheck-quirk;
+			status = "disabled";
+		};
+	};
+
 	gic: interrupt-controller@ff811000 {
 		compatible = "arm,gic-400";
 		#interrupt-cells = <3>;

commit 80d50029ddc853cbb443cf16a8983c951c26bfa2
Author: Cameron Nemo <cnemo@tutanota.com>
Date:   Sat Aug 15 13:09:36 2020 -0700

    usb: dwc3: of-simple: Add compatible for rk3328
    
    Add a compatible to be hooked into by the Rockchip rk3328 device tree.
    
    The rk3399 compatible cannot be reused because the rk3328 SoCs may
    require a specialized driver in the future, and old device trees must
    remain compatible with newer kernels.
    
    Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

diff --git a/drivers/usb/dwc3/dwc3-of-simple.c b/drivers/usb/dwc3/dwc3-of-simple.c
index e62ecd22b3ed..93bc343287d3 100644
--- a/drivers/usb/dwc3/dwc3-of-simple.c
+++ b/drivers/usb/dwc3/dwc3-of-simple.c
@@ -171,6 +171,7 @@ static const struct dev_pm_ops dwc3_of_simple_dev_pm_ops = {
 };
 
 static const struct of_device_id of_dwc3_simple_match[] = {
+	{ .compatible = "rockchip,rk3328-dwc3" },
 	{ .compatible = "rockchip,rk3399-dwc3" },
 	{ .compatible = "xlnx,zynqmp-dwc3" },
 	{ .compatible = "cavium,octeon-7130-usb-uctl" },

commit 3cf988130e1ef4ba148fe6169f4f3da933b9274c
Author: Cameron Nemo <cnemo@tutanota.com>
Date:   Sun Aug 16 15:21:32 2020 -0700

    dt-bindings: usb: add rk3328 dwc3 docs
    
    Document compatible for dwc3 on the Rockchip rk3328 platform.
    
    Signed-off-by: Cameron Nemo <cnemo@tutanota.com>

diff --git a/Documentation/devicetree/bindings/usb/dwc3.txt b/Documentation/devicetree/bindings/usb/dwc3.txt
index 1aae2b6160c1..f9a70d48802e 100644
--- a/Documentation/devicetree/bindings/usb/dwc3.txt
+++ b/Documentation/devicetree/bindings/usb/dwc3.txt
@@ -25,6 +25,7 @@ Exception for clocks:
     "ti,am437x-dwc3"
     "ti,dwc3"
     "ti,keystone-dwc3"
+    "rockchip,rk3328-dwc3"
     "rockchip,rk3399-dwc3"
     "xlnx,zynqmp-dwc3"
 
diff --git a/Documentation/devicetree/bindings/usb/rockchip,dwc3.txt b/Documentation/devicetree/bindings/usb/rockchip,dwc3.txt
index 94520493233b..b41f30a61be6 100644
--- a/Documentation/devicetree/bindings/usb/rockchip,dwc3.txt
+++ b/Documentation/devicetree/bindings/usb/rockchip,dwc3.txt
@@ -1,7 +1,8 @@
 Rockchip SuperSpeed DWC3 USB SoC controller
 
 Required properties:
-- compatible:	should contain "rockchip,rk3399-dwc3" for rk3399 SoC
+- compatible:	should contain "rockchip,rk3328-dwc3" for rk3328 SoC
+		or "rockchip,rk3399-dwc3" for rk3399 SoC
 - clocks:	A list of phandle + clock-specifier pairs for the
 		clocks listed in clock-names
 - clock-names:	Should contain the following:
